### PR TITLE
Feature/api integration

### PR DESCRIPTION
**Pull request by HenriChabert**
Big PR as repo has been created too late, won't happen again.

### CODE REVIEW

File: python-connectors/open-weather-map_once-call-data/connector.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+class OpenWeatherMapConnector(Connector):
+
+    def __init__(self, config, plugin_config):
+        Connector.__init__(self, config, plugin_config)
+
+        preset_config = self.config.get("preset_config")
+
+        self.cache_location = get_cache_location_from_configs(
+            cache_location=self.plugin_config.get('cache_location'),
```
Comment (MayeulRousselet) [inline python-connectors/open-weather-map_once-call-data/connector.py]: Be consistent with the quotes (use double quotes for all strings)

File: python-lib/openweathermap_utils/openWeatherMapAPI.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        self.api_version = '2.5'
+        self.base_url = f'https://api.openweathermap.org/data/{self.api_version}/'
+        self.datetime_schema = '%Y-%m-%d'
+        self.available_columns = constants.COL_TYPES
+        self.cache = cache
+        self.api_calls_nb = 0
+
+    def _get_query(self, endpoint, params):
+        """
+        Realises an HTTP GET query on the endpoint using Requests library.
```
Comment (MayeulRousselet) [inline python-lib/openweathermap_utils/openWeatherMapAPI.py]: REplace Realises by perform

File: python-lib/openweathermap_utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+def log_txt(txt):
+    output_txt = '\n'
+    output_txt += log_sep()
+    output_txt += txt
+    output_txt += log_sep()
+    return output_txt
+
+
+def debug_func(function):
```
Comment (MayeulRousselet) [inline python-lib/openweathermap_utils/utils.py]: remove :)

File: python-lib/openweathermap_utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    if type_ == 'float':
+        return float(value)
+    if type_ == 'int':
+        return int(value)
+    if type_ == 'boolean':
+        return value in ['true', True, 'T', 1, 'Vrai', 'V', '1', 'True']
+    return str(value)
+
+
+def dbg_msg(msg, title=""):
```
Comment (MayeulRousselet) [inline python-lib/openweathermap_utils/utils.py]: not called, to be removed

File: python-lib/openweathermap_utils/utils.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            return res, OpenWeatherMapAPIError(status_code=200, text='')
+        except OpenWeatherMapAPIError as owm_error:
+            return [{}], owm_error
+        except HTTPError as http_error:
+            owm_error = OpenWeatherMapAPIError(
+                http_error.response.text,
+                status_code=int(http_error.response.status_code),
+                text=http_error.response.text
+            )
+            if owm_error.status_code == 401:
```
Comment (MayeulRousselet) [inline python-lib/openweathermap_utils/utils.py]: What about 403?

File: python-lib/openweathermap_utils/utils.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        for k, v in el.items():
+            flatten_dict_rec(v, names + [str(k)], new_dict)
+    elif isinstance(el, list):
+        for i, e in enumerate(el):
+            flatten_dict_rec(e, names + [str(i)], new_dict)
+    else:
+        new_dict['.'.join(names)] = el
+
+
+def flatten_dict(d):
```
Comment (MayeulRousselet) [inline python-lib/openweathermap_utils/utils.py:66]: Need a comment to explain what this method does

**Review by MayeulRousselet — CHANGES_REQUESTED** [review]
Very very good work.
Question: How does a user can clear the cache?

File: python-connectors/open-weather-map_once-call-data/connector.py (hunk 1 above)
Comment (HenriChabert) [reply python-connectors/open-weather-map_once-call-data/connector.py]: Definitely right, fixed in d49b700

File: python-lib/openweathermap_utils/openWeatherMapAPI.py (hunk 2 above)
Comment (HenriChabert) [reply python-lib/openweathermap_utils/openWeatherMapAPI.py]: Fixed in d3e0af3

File: python-lib/openweathermap_utils/utils.py (hunk 3 above)
Comment (HenriChabert) [reply python-lib/openweathermap_utils/utils.py]: Fixed in 7227f75

File: python-lib/openweathermap_utils/utils.py (hunk 4 above)
Comment (HenriChabert) [reply python-lib/openweathermap_utils/utils.py]: Fixed in 7227f75

File: python-lib/openweathermap_utils/utils.py (hunk 5 above)
Comment (HenriChabert) [reply python-lib/openweathermap_utils/utils.py]: OpenWeatherMap raises a 401 error when the API key is not authorized to call a service. I don't know if it ever raises a 403 but added it in d3e0af3

File: python-lib/openweathermap_utils/utils.py (hunk 6 above)
Comment (HenriChabert) [reply python-lib/openweathermap_utils/utils.py:66]: Fixed in 93397f3

File: python-lib/openweathermap_utils/utils.py (hunk 5 above)
Comment (MayeulRousselet) [reply python-lib/openweathermap_utils/utils.py]: Ok, don't need to add a 403 so.

**Review by MayeulRousselet — APPROVED** [review]

**Pull request merged**